src/uct/ib/ud/base/ud_ep.c

```diff
@@ -828,6 +834,16 @@ void uct_ud_ep_process_rx(uct_ud_iface_t *iface, uct_ud_neth_t *neth, unsigned b
                        ep->rx.ooo_pkts.head_sn, neth->psn);
     }
 
+    if (ucs_unlikely(neth->packet_type & UCT_UD_PACKET_FLAG_NAK)) {
+        ep->tx.naked = 1;
+        goto out;
+    }
```

Comment:
can we add under if in line 847 for opt? are nak packets always not active messages?